src/plugins/dashboard/public/dashboard_container/embeddable/api/overlays/save_modal.tsx

```diff
@@ -36,65 +36,61 @@ interface Props {
   tags?: string[];
   timeRestore: boolean;
   showCopyOnSave: boolean;
+  showStoreTimeOnSave?: boolean;
+  customModalTitle?: string;
 }
 
-interface State {
-  tags: string[];
-  timeRestore: boolean;
-}
+type SaveDashboardHandler = (args: {
+  newTitle: string;
+  newDescription: string;
+  newCopyOnSave: boolean;
+  isTitleDuplicateConfirmed: boolean;
+  onTitleDuplicate: () => void;
+}) => ReturnType<DashboardSaveModalProps['onSave']>;
 
-export class DashboardSaveModal extends React.Component<Props, State> {
-  state: State = {
-    timeRestore: this.props.timeRestore,
-    tags: this.props.tags ?? [],
-  };
+export const DashboardSaveModal: React.FC<DashboardSaveModalProps> = ({
+  customModalTitle,
+  description,
+  onClose,
+  onSave,
+  showCopyOnSave,
+  showStoreTimeOnSave = true,
+  tags,
+  title,
+  timeRestore,
+}) => {
+  const [selectedTags, setSelectedTags] = React.useState<string[]>(tags ?? []);
+  const [persistSelectedTimeInterval, setPersistSelectedTimeInterval] = React.useState(timeRestore);
 
-  constructor(props: Props) {
-    super(props);
-  }
-
-  saveDashboard = ({
-    newTitle,
-    newDescription,
-    newCopyOnSave,
-    isTitleDuplicateConfirmed,
-    onTitleDuplicate,
-  }: {
-    newTitle: string;
-    newDescription: string;
-    newCopyOnSave: boolean;
-    isTitleDuplicateConfirmed: boolean;
-    onTitleDuplicate: () => void;
-  }) => {
-    this.props.onSave({
-      newTitle,
-      newDescription,
-      newCopyOnSave,
-      newTimeRestore: this.state.timeRestore,
-      isTitleDuplicateConfirmed,
-      onTitleDuplicate,
-      newTags: this.state.tags,
-    });
-  };
+  const saveDashboard = React.useCallback<SaveDashboardHandler>(
```

Comment:
nit: our team usually de-structures react imports 
```suggestion
  const saveDashboard = useCallback<SaveDashboardHandler>(
```